packages/blocks/src/api/utils.js

```diff
@@ -144,7 +145,7 @@ export function getBlockLabel( blockType, attributes, context = 'visual' ) {
 	// Attempt to find entity title if block is a template part.
 	// Require slug to request, otherwise entity is uncreated and will throw 404.
 	if ( 'core/template-part' === blockType.name && attributes.slug ) {
-		const entity = select( 'core' ).getEntityRecord(
+		const entity = select( coreStore ).getEntityRecord(
```

Comment:
Yep, this should definitely be moved to the block itself :) I think we have a function for that right?

Reply:
Yes, I'm on it. I found the PR where it was introduced: https://github.com/WordPress/gutenberg/pull/28330. It only confirms that it should be moved.